src/rules/no-cycle.js

```diff
@@ -10,6 +10,16 @@ import moduleVisitor, { makeOptionsSchema } from 'eslint-module-utils/moduleVisi
 import docsUrl from '../docsUrl';
 
 const traversed = new Set();
+/** 
+ * Cache of resolved paths for faster subsequent lookups
+ * @type {{ [importPath: string]: string }} 
+ */
+const resolvedPaths = {};
+/** 
+ * Cache of external module evaluations for faster subsequent lookups
+ * @type {{ [resolvedPath: string]: boolean }} 
+ */
+const externalModules = {};
 
```

Comment:
these caches need to be cleared in the same place `traversed` is

Reply:
The `Program:exit` hook is called at the exit of each file, so clearing the caches there cut the number of cache hits in our repo by about 50% (although it only added ~45s to our repo test case)

The high level attempt at this update was to quick filter down to "maybe external" modules like `'react'`, `'@reduxjs/toolkit'` or `'@/some/alias/path'` - where those resolved paths can be safely cached across files 🤔 

Is that possible to accomplish with any of the existing data structures?

Reply:
oh, hmm - you're right, it's less about clearing it after the file and more about clearing it after the "run" - since in editors, eslint can be long-running.

I think that in every file, there could be a different resolver, so the cache would have to be on the resolver *and* the resolver config *and*  the fully normalized file path, i think?